### ISSUE TITLE
Error handling constant value in aggregation function

### ISSUE THREAD

**Issue by Novemser** [pingcap/tispark#49]
If we pass constant characters as parameter to aggregation function like ```sum() avg()``` etc.,  spark will encounter unexpected problems.

For query ```spark.sql("select avg('some_random_word') from customer").show```, spark throw exceptions as below:

```
17/10/10 17:34:48 ERROR Executor: Exception in task 1.0 in stage 20.0 (TID 93)
java.lang.ArrayIndexOutOfBoundsException: 1
	at com.pingcap.tikv.row.ObjectRowImpl.set(ObjectRowImpl.java:155)
	at com.pingcap.tikv.operation.transformer.Cast.set(Cast.java:33)
	at com.pingcap.tikv.operation.transformer.RowTransformer.transform(RowTransformer.java:106)
	at com.pingcap.tispark.TiRDD$$anon$1.toSparkRow(TiRDD.scala:58)
	at com.pingcap.tispark.TiRDD$$anon$1.next(TiRDD.scala:70)
	at com.pingcap.tispark.TiRDD$$anon$1.next(TiRDD.scala:50)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIterator.agg_doAggregateWithoutKey$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIterator.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anonfun$8$$anon$1.hasNext(WholeStageCodegenExec.scala:395)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:125)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:96)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:335)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
17/10/10 17:34:48 WARN TaskSetManager: Lost task 1.0 in stage 20.0 (TID 93, localhost, executor driver): java.lang.ArrayIndexOutOfBoundsException: 1
	at com.pingcap.tikv.row.ObjectRowImpl.set(ObjectRowImpl.java:155)
	at com.pingcap.tikv.operation.transformer.Cast.set(Cast.java:33)
	at com.pingcap.tikv.operation.transformer.RowTransformer.transform(RowTransformer.java:106)
	at com.pingcap.tispark.TiRDD$$anon$1.toSparkRow(TiRDD.scala:58)
	at com.pingcap.tispark.TiRDD$$anon$1.next(TiRDD.scala:70)
	at com.pingcap.tispark.TiRDD$$anon$1.next(TiRDD.scala:50)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIterator.agg_doAggregateWithoutKey$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIterator.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anonfun$8$$anon$1.hasNext(WholeStageCodegenExec.scala:395)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:125)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:96)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:335)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

``` 

Please

**Comment by Novemser** [comment]
Known affected version:
```
pre-dag-100-gf550205-CLI-pre-dag-64-g75599c1
```

**Comment by Novemser** [comment]
Should be fixed by tikv-client: [#183](https://github.com/pingcap/tikv-client-lib-java/pull/183)
